atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/ValidatingQueryRewritingKeyValueService.java

```diff
@@ -112,6 +113,15 @@ public void delete(TableReference tableRef, Multimap<Cell, Long> keys) {
     }
 
     @Override
+    public void deleteRange(TableReference tableRef, RangeRequest rangeRequest) {
+        if (!rangeRequest.getColumnNames().isEmpty()) {
```

Comment:
Yeah, it would be something of a breaking change to introduce this validation where it was not previously there. I can see that causing an issue somewhere - might be worth raising the idea at a future group check-in.